bindings/node/test/clientEncryption.test.js

```diff
@@ -815,4 +829,115 @@ describe('ClientEncryption', function () {
   it('should provide the libmongocrypt version', function () {
     expect(ClientEncryption.libmongocryptVersion).to.be.a('string');
   });
+
+  describe('createEncryptedCollection()', () => {
+    /** @type {InstanceType<ClientEncryption>} */
+    const client = new MockClient();
+    let clientEncryption;
+    let db;
+    const collectionName = 'secure';
+
+    beforeEach(async function () {
+      clientEncryption = new ClientEncryption(client, {
+        keyVaultNamespace: 'client.encryption',
+        kmsProviders: { local: { key: Buffer.alloc(96, 0) } }
+      });
+
+      db = client.db('createEncryptedCollectionDb');
+      await db.dropDatabase().catch(() => null);
+    });
+
+    afterEach(async () => {
+      sinon.restore();
+      await db.dropDatabase().catch(() => null);
+    });
+
+    it('throws if options are omitted', async () => {
```

Comment:
> destructuring errors are opaque

It actually states the name of the missing property
```txt
> createEncryptedCollection(db, '')
TypeError: Cannot destructure property 'provider' of 'undefined' as it is undefined.
```

> throwing if it is not provided ... are we testing this only because we're destructing options from it? it does feel odd to test for the presence of only 1/3 required arguments

Yea so we generally don't try to call our functions with missing arguments unless there is overloads that permit it. For example, `insertOne` crashes with `Cannot read properties of undefined` if you don't pass it an object to insert. I added this test because it is a new way that we just let the runtime handle crashing, but I think its equivalent to that approach we've taken so far, but with a _slightly_ enhanced error message.

> missing a test for options.provider

this will be validated by `createDataKey` in the c++

Reply:
A destructing error tells you what key was undefined, but not on what object it was accessed:

```
function bar({ name })  {
	console.log(name);
}

bar();

// TypeError: Cannot destructure property 'name' of 'undefined' as it is undefined.
```

An error that says something like "option `<optionName>` is required but was not provided` is much clearer and more actionable.

Reply:
We discussed this and I think landed on being happy with the errors that come from destructuring. These errors are only encountered when folks "break" the TS contracts we've setup.